packages/content/src/portal.ts

```diff
@@ -16,10 +16,61 @@ import {
   getItemDataUrl,
   getItemThumbnailUrl,
   cloneObject,
-  includes
+  includes,
+  failSafe
 } from "@esri/hub-common";
 import { getContentMetadata } from "./metadata";
 
+function getGroupIds(
+  itemId: string,
+  requestOptions: IHubRequestOptions
+): Promise<string[]> {
+  return getItemGroups(itemId, requestOptions).then(response => {
+    const { admin, member, other } = response;
+    return [...admin, ...member, ...other].map(group => group.id);
+  });
+}
+
+// simultaneously execute multiple async requests to populate content properties
+// returns a hash of all the resolved property values that also has
+// an array of property names for which the request failed
+// NOTE: assumes each request takes only the item id and request options
+function fetchProperties(
+  propertyRequests: {
+    [key: string]: (
+      itemId: string,
+      requestOptions: IHubRequestOptions
+    ) => Promise<unknown>;
+  },
+  itemId: string,
+  requestOptions: IHubRequestOptions
+): Promise<{ [key: string]: unknown }> {
+  const propertyNames = Object.keys(propertyRequests);
+  // ideally we'd be able to use hashSettled(), but that's not standard
+  // so instead return the property name to indicate that the request failed
+  const requests = propertyNames.map(propertyName =>
+    failSafe(propertyRequests[propertyName], propertyName)(
+      itemId,
+      requestOptions
+    )
+  );
+  return Promise.all(requests).then(values => {
+    return values.reduce(
+      (properties, value, i) => {
+        const propertyName = propertyNames[i];
+        if (value === propertyName) {
+          // capture that the request failed
+          properties.errors.push(propertyName);
+        } else {
+          properties[propertyName] = value;
+        }
+        return properties;
+      },
+      { errors: [] } as { [key: string]: unknown }
```

Comment:
leaving this undocumented for now until I verify the format of the errors returned by the Hub REST API

Reply:
Leave a `TODO` comment to this effect perhaps?